configure: error: libmilter not found

I have libmilter installed in a location that is most convenient to me, let say /opt/libmilter. Opendkim is configured "--with-milter=/opt/libmilter" and the compiler is supposed to have an easy life with the following:

```
export CPPFLAGS="$CPPFLAGS -I/opt/libmilter/include"
export LDFLAGS="$LDFLAGS -L/opt/libmilter/lib"
```
Now, the library is there, opendkim is configured pointing its nose to where the library IS, and yet, it fails to find it. 

Configure happens to be 24232 lines long, generated as follows:

```
   libtoolize -v -i --copy --force
   aclocal -I m4 -Wall
   autoheader -Wall -v

   autoconf -Wall -v
   # configure.ac:68: warning: The macro `AC_PROG_LIBTOOL' is obsolete.
   # configure.ac:2716: You should run autoupdate.

   autoupdate

   automake --add-missing --force-missing --foreign --copy --warnings=override -Wall
```

This is the test that failed:

```
    if test x"$LIBMILTER_LIBDIRS" = x""
        then
                as_fn_error $? "libmilter not found" "$LINENO" 5
        fi
```

where $LIBMILTER_LIBDIRS is defined as follows:

```
if test "$ac_res" != no; then :
  test "$ac_res" = "none required" || LIBS="$ac_res $LIBS"

                                LIBMILTER_LIBDIRS="-L$milterpath/$d"
                                LIBMILTER_LIBS="-lmilter"
                                breakloop="yes"

fi
```
I commented the conditional, forcing the assignment, and the test passed. 

Configure went on with the next test, breaking at 

`configure: error: libldns not found in lib`

So, the block below is failing, multiple times. 

```
if test "$ac_res" != no; then :
  test "$ac_res" = "none required" || LIBS="$ac_res $LIBS"
```

--- a/configure.ac
+++ b/configure.ac
@@ -963,6 +963,7 @@ then
 		fi
 	done
 fi
+milter_explicit="no"
 case "$milterpath" in
 	no)
 		if test x"$enable_filter" = x"yes"
@@ -980,6 +981,7 @@ case "$milterpath" in
 			AC_MSG_ERROR([milter includes not found at $milterpath])
 		fi
 		AC_MSG_RESULT([$milterpath])
+		milter_explicit="yes"
 		;;
 	*)
 		AC_MSG_ERROR([milter not found])
@@ -1001,52 +1003,62 @@ then
 	saved_LIBS="$LIBS"
 
 	CC="$PTHREAD_CC"
-	LIBS="$outer_LIBS $PTHREAD_LIBS $saved_LIBS"
 	CPPFLAGS="$LIBMILTER_INCDIRS $saved_CPPFLAGS"
 	CFLAGS="$PTHREAD_CFLAGS $saved_CFLAGS"
-	LDFLAGS="$outer_LDFLAGS $PTHREAD_CFLAGS $saved_LDFLAGS"
 
-	breakloop="no"
-	for d in lib lib64 lib/libmilter
-	do
-		unset ac_cv_search_smfi_register
-		LDFLAGS="$outer_LDFLAGS $PTHREAD_CFLAGS -L$milterpath/$d $saved_LDFLAGS"
-		AC_SEARCH_LIBS([smfi_register], [milter],
-		               [
-		               	LIBMILTER_LIBDIRS="-L$milterpath/$d"
-		               	LIBMILTER_LIBS="-lmilter"
-		               	breakloop="yes"
-		               ])
-
-		AC_CHECK_FUNC([smfi_insheader],
-			      AC_DEFINE([HAVE_SMFI_INSHEADER], 1,
-					[Define if libmilter has smfi_insheader()]))
-
-		AC_CHECK_FUNC([smfi_opensocket],
-			      AC_DEFINE([HAVE_SMFI_OPENSOCKET], 1,
-					[Define if libmilter has smfi_opensocket()]))
-
-		AC_CHECK_FUNC([smfi_progress],
-			      AC_DEFINE([HAVE_SMFI_PROGRESS], 1,
-					[Define if libmilter has smfi_progress()]))
-
-		AC_CHECK_FUNC([smfi_setsymlist],
-			      AC_DEFINE([HAVE_SMFI_SETSYMLIST], 1,
-					[Define if libmilter has smfi_setsymlist()]))
-
-		AC_CHECK_FUNC([smfi_version],
-			      AC_DEFINE([HAVE_SMFI_VERSION], 1,
-					[Define if libmilter has smfi_version()]))
-
-		if test x"$breakloop" = x"yes"
-		then
-			break
-		fi
-	done
-	if test x"$LIBMILTER_LIBDIRS" = x""
+	if test x"$milter_explicit" = x"yes"
 	then
-		AC_MSG_ERROR([libmilter not found])
+		LIBMILTER_LIBDIRS="-L$milterpath/lib"
+		LIBMILTER_LIBS="-lmilter"
+		LDFLAGS="$outer_LDFLAGS $PTHREAD_CFLAGS $LIBMILTER_LIBDIRS $saved_LDFLAGS"
+		LIBS="$outer_LIBS $PTHREAD_LIBS $LIBMILTER_LIBS $saved_LIBS"
+	else
+		LIBS="$outer_LIBS $PTHREAD_LIBS $saved_LIBS"
+		LDFLAGS="$outer_LDFLAGS $PTHREAD_CFLAGS $saved_LDFLAGS"
+
+		breakloop="no"
+		for d in lib lib64 lib/libmilter
+		do
+			unset ac_cv_search_smfi_register
+			LDFLAGS="$outer_LDFLAGS $PTHREAD_CFLAGS -L$milterpath/$d $saved_LDFLAGS"
+			AC_SEARCH_LIBS([smfi_register], [milter],
+			               [
+			               	LIBMILTER_LIBDIRS="-L$milterpath/$d"
+			               	LIBMILTER_LIBS="-lmilter"
+			               	breakloop="yes"
+			               ])
+			if test x"$breakloop" = x"yes"
+			then
+				break
+			fi
+		done
+		if test x"$LIBMILTER_LIBDIRS" = x""
+		then
+			AC_MSG_ERROR([libmilter not found])
+		fi
+		LDFLAGS="$outer_LDFLAGS $PTHREAD_CFLAGS $LIBMILTER_LIBDIRS $saved_LDFLAGS"
+		LIBS="$outer_LIBS $PTHREAD_LIBS $LIBMILTER_LIBS $saved_LIBS"
 	fi
+
+	AC_CHECK_FUNC([smfi_insheader],
+		      AC_DEFINE([HAVE_SMFI_INSHEADER], 1,
+				[Define if libmilter has smfi_insheader()]))
+
+	AC_CHECK_FUNC([smfi_opensocket],
+		      AC_DEFINE([HAVE_SMFI_OPENSOCKET], 1,
+				[Define if libmilter has smfi_opensocket()]))
+
+	AC_CHECK_FUNC([smfi_progress],
+		      AC_DEFINE([HAVE_SMFI_PROGRESS], 1,
+				[Define if libmilter has smfi_progress()]))
+
+	AC_CHECK_FUNC([smfi_setsymlist],
+		      AC_DEFINE([HAVE_SMFI_SETSYMLIST], 1,
+				[Define if libmilter has smfi_setsymlist()]))
+
+	AC_CHECK_FUNC([smfi_version],
+		      AC_DEFINE([HAVE_SMFI_VERSION], 1,
+				[Define if libmilter has smfi_version()]))
 
 	CC="$saved_CC"
 	CPPFLAGS="$saved_CPPFLAGS"

